src/benchmark/file_io_read_micro_benchmark.cpp

```diff
@@ -406,25 +378,20 @@ void FileIOMicroReadBenchmarkFixture::aio_single_threaded(benchmark::State& stat
     read_data.resize(NUMBER_OF_ELEMENTS);
     state.ResumeTiming();
 
-    struct aiocb aiocb;
+    struct aiocb requests[1];
+    const struct aiocb * aio_list[1]; //aio_suspend expects a list of async I/O requests
 
-    memset(&aiocb, 0, sizeof(struct aiocb));
-    aiocb.aio_fildes = fd;
-    aiocb.aio_offset = 0;
-    aiocb.aio_buf = std::data(read_data);
-    aiocb.aio_nbytes = NUMBER_OF_BYTES;
-    aiocb.aio_lio_opcode = LIO_READ;
+    create_aio_request(requests[0], fd, 0, std::data(read_data), NUMBER_OF_BYTES, LIO_READ);
+    aio_list[0] = &requests[0];
 
-    Assert(aio_read(&aiocb) == 0, "Read error: " + strerror(errno));
+    Assert(aio_read(&requests[0]) == 0, "Read error: " + strerror(errno));
 
-    /* Wait until end of transaction */
-    auto err = int{0};
-    while ((err = aio_error(&aiocb)) == EINPROGRESS);
+    //suspend this thread until the async I/O request has completed
+    aio_suspend(aio_list,1,NULL);
```

Comment:
The way I understand the doc (and by the name :D) this suspends the thread to the OS. This suspension costs ~50mus. The thread will be awakened when the IO is finished. For modern SSDs, suspension should not yield any advantages since the IO is already as fast as just putting a thread to sleep.

I think for a fair comparison (other methods block until I/O is done), your implementation should busy loop for the execution of the request.